### PR TITLE
Let proton subscribe to bucketspaces config.

### PR DESCRIPTION
**Pull request by toregge**
@geirst, @vekterli: please review

### CODE REVIEW

**Review by geirst — APPROVED** [review]
👍

File: searchcore/src/vespa/searchcore/proton/server/bootstrapconfig.cpp [hunk 1]
```diff
@@ -56,7 +61,9 @@ BootstrapConfig::valid() const
     return _documenttypes.get() != NULL &&
                     _repo.get() != NULL &&
                   _proton.get() != NULL &&
-      _tuneFileDocumentDB.get() != NULL;
+        _fileDistributorRpc.get() != NULL &&
+        _bucketspaces.get() != NULL &&
+        _tuneFileDocumentDB.get() != NULL;
```
Comment (vekterli) [inline searchcore/src/vespa/searchcore/proton/server/bootstrapconfig.cpp]: These can be upgraded to `nullptr`

**Review by vekterli — APPROVED** [review]
:+1:

File: searchcore/src/vespa/searchcore/proton/server/bootstrapconfig.cpp (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/bootstrapconfig.cpp]: I guess even better to just use the boolean operators and just remove the whole ".get() != NULL"

Comment (toregge) [reply searchcore/src/vespa/searchcore/proton/server/bootstrapconfig.cpp]: This is addressed in PR #4599

**Pull request merged**